// @ant-design/pro-editor/es/SortableList/container/index.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["SHOW_STORE_IN_DEVTOOLS", "className", "style"];
import { forwardRef, memo } from 'react';
import { ConfigProvider } from "../../ConfigProvider";
import App from "./App";
import { SortableListProvider } from "./Provider";
import StoreUpdater from "./StoreUpdater";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export { SortableListProvider } from "./Provider";
export var SortableList = /*#__PURE__*/memo( /*#__PURE__*/forwardRef(function (props, ref) {
  var SHOW_STORE_IN_DEVTOOLS = props.SHOW_STORE_IN_DEVTOOLS,
    className = props.className,
    style = props.style,
    res = _objectWithoutProperties(props, _excluded);
  return /*#__PURE__*/_jsx(ConfigProvider, {
    children: /*#__PURE__*/_jsxs(SortableListProvider, {
      showDevtools: SHOW_STORE_IN_DEVTOOLS,
      children: [/*#__PURE__*/_jsx(App, {
        style: style,
        className: className
      }), /*#__PURE__*/_jsx(StoreUpdater, _objectSpread(_objectSpread({}, res), {}, {
        ref: ref
      }))]
    })
  });
}));
