// @ant-design/pro-editor/es/SortableTree/hooks/useSortableTree.js
import { useMemoizedFn } from 'ahooks';
import { useMemo } from 'react';
import { dataFlattenSelector, useStoreApi } from "../store";

/**
 * SortableTree 实例对象
 * @template T 节点数据类型
 */

export var useSortableTree = function useSortableTree() {
  var storeApi = useStoreApi();
  var _storeApi$getState = storeApi.getState(),
    dispatchTreeData = _storeApi$getState.dispatchTreeData,
    removeNode = _storeApi$getState.removeNode,
    addNode = _storeApi$getState.addNode,
    toggleNode = _storeApi$getState.toggleNode,
    updateNodeContent = _storeApi$getState.updateNodeContent,
    toggleExtraVisible = _storeApi$getState.toggleExtraVisible,
    deselectedAll = _storeApi$getState.deselectedAll,
    selectedNode = _storeApi$getState.selectedNode;
  var getActiveId = useMemoizedFn(function () {
    return storeApi.getState().activeId;
  });
  var getTreeData = useMemoizedFn(function () {
    return storeApi.getState().treeData;
  });
  var getFlattenData = useMemoizedFn(function () {
    return dataFlattenSelector(storeApi.getState());
  });
  var getSelectedIds = useMemoizedFn(function () {
    return storeApi.getState().selectedIds;
  });
  return useMemo(function () {
    return {
      dispatchTreeData: dispatchTreeData,
      toggleExtraVisible: toggleExtraVisible,
      removeNode: removeNode,
      deselectedAll: deselectedAll,
      addNode: addNode,
      selectedNode: selectedNode,
      toggleNode: toggleNode,
      updateNodeContent: updateNodeContent,
      getActiveId: getActiveId,
      getTreeData: getTreeData,
      getFlattenData: getFlattenData,
      getSelectedIds: getSelectedIds
    };
  }, []);
};
