// @ant-design/pro-editor/es/ProBuilder/hooks/useProBuilder.js
import { useMemoizedFn } from 'ahooks';
import { useMemo } from 'react';
import { useStoreApi } from "../store";

/**
 * ProBuilder 实例对象
 * @template Config - 配置信息的类型
 * @template Props - 组件属性的类型
 */

export var useProBuilder = function useProBuilder() {
  var storeApi = useStoreApi();
  var _storeApi$getState = storeApi.getState(),
    deselectCanvas = _storeApi$getState.deselectCanvas,
    undoStack = _storeApi$getState.undoStack,
    redoStack = _storeApi$getState.redoStack,
    setConfig = _storeApi$getState.setConfig,
    exportConfig = _storeApi$getState.exportConfig,
    setViewport = _storeApi$getState.setViewport,
    resetConfig = _storeApi$getState.resetConfig,
    undo = _storeApi$getState.undo,
    redo = _storeApi$getState.redo,
    setCanvasInteraction = _storeApi$getState.setCanvasInteraction,
    togglePanelExpand = _storeApi$getState.togglePanelExpand;
  var getViewport = useMemoizedFn(function () {
    return storeApi.getState().editorAwareness.viewport;
  });
  var getEditorAwareness = useMemoizedFn(function () {
    return storeApi.getState().editorAwareness;
  });
  var getConfig = useMemoizedFn(function () {
    return storeApi.getState().config;
  });
  var getProps = useMemoizedFn(function () {
    return storeApi.getState().props;
  });
  var getCanvasInteraction = useMemoizedFn(function () {
    return storeApi.getState().interaction;
  });
  return useMemo(function () {
    return {
      getConfig: getConfig,
      setConfig: setConfig,
      exportConfig: exportConfig,
      resetConfig: resetConfig,
      getViewport: getViewport,
      setViewport: setViewport,
      getCanvasInteraction: getCanvasInteraction,
      setCanvasInteraction: setCanvasInteraction,
      deselectCanvas: deselectCanvas,
      undo: undo,
      redo: redo,
      undoStack: undoStack,
      redoStack: redoStack,
      getProps: getProps,
      getEditorAwareness: getEditorAwareness,
      togglePanelExpand: togglePanelExpand
    };
  }, []);
};
