// @ant-design/pro-editor/es/Awareness/Avatars/Avatar.js
import { Avatar as A, Badge, Tooltip } from 'antd';
import Color from 'color';
import { memo } from 'react';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var Avatar = /*#__PURE__*/memo(function (_ref) {
  var name = _ref.name,
    color = _ref.color,
    onClick = _ref.onClick,
    active = _ref.active,
    current = _ref.current,
    following = _ref.following;
  if (!name) return /*#__PURE__*/_jsx(A, {});
  var colorModel = Color(color);
  return /*#__PURE__*/_jsx(Tooltip, {
    title: name,
    showArrow: false,
    children: /*#__PURE__*/_jsxs(A, {
      shape: 'circle',
      style: {
        background: color,
        outline: following ? "2px solid ".concat(color) : '',
        color: colorModel.isLight() ? 'black' : 'inherit',
        zIndex: following ? 1000 : active ? 100 : 0,
        filter: active ? 'none' : 'grayscale(80%)',
        overflow: 'initial',
        cursor: current ? 'inherit' : 'pointer'
      },
      onClick: onClick,
      children: [name.slice(0, 1), /*#__PURE__*/_jsx(Badge, {
        status: active ? 'success' : 'default',
        color: active ? undefined : '#d9d9d9',
        style: {
          position: 'absolute',
          left: 11,
          top: 11
        }
      })]
    })
  });
});
export default Avatar;
