// @ant-design/pro-editor/es/ColumnList/ColumnItem.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _taggedTemplateLiteral from "@babel/runtime/helpers/esm/taggedTemplateLiteral";
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5;
import { DeleteAction, HandleAction, useSortableList } from "./..";
import { createStyles } from 'antd-style';
import classNames from 'classnames';
import { memo } from 'react';
import { Flexbox } from 'react-layout-kit';
import ControlInput from "./renderItem/Input";
import ControlSelect from "./renderItem/Select";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var useStyle = createStyles(function (_ref) {
  var css = _ref.css,
    cx = _ref.cx,
    token = _ref.token,
    prefixCls = _ref.prefixCls;
  var prefix = "".concat(prefixCls, "-").concat(token.editorPrefix, "-item");
  return {
    item: cx(prefix, css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n        position: relative;\n        width: 100%;\n\n        &:hover .", "-actions {\n          opacity: 1;\n        }\n      "])), prefix)),
    content: cx("".concat(prefix, "-content"), css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n        flex: 1;\n        font-size: 12px;\n        border-radius: 2px;\n        min-width: 48px;\n      "])))),
    actions: cx("".concat(prefix, "-actions"), css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n        z-index: 10;\n        color: hsla(0, 0, 0, 0.45);\n        opacity: 0;\n      "])))),
    handle: cx("".concat(prefix, "-handle"), css(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n        position: absolute;\n        left: 0;\n      "])))),
    actionsRight: cx("".concat(prefix, "-actions-right"), css(_templateObject5 || (_templateObject5 = _taggedTemplateLiteral(["\n        position: absolute;\n        top: 1px;\n        right: 1px;\n        align-self: flex-end;\n        overflow: hidden;\n        border-radius: 1px;\n        // \u91C7\u7528\u80CC\u666F\u6A21\u7CCA\u6765\u89E3\u51B3\u591A\u79CD\u80CC\u666F\u8272\u4E0B\u8986\u76D6\u5185\u5BB9\u7684\u95EE\u9898 TODO\uFF1AFireFox \u517C\u5BB9\n        backdrop-filter: blur(5px);\n        //background-image: linear-gradient(90deg, rgba(255, 255, 255, 0) 0%, @bg-2 10%, @bg-2 100%);\n      "]))))
  };
});
var ColumnItem = /*#__PURE__*/memo(function (_ref2) {
  var item = _ref2.item,
    index = _ref2.index,
    columns = _ref2.columns,
    listeners = _ref2.listeners,
    actions = _ref2.actions,
    hideRemove = _ref2.hideRemove,
    dragging = _ref2.dragging,
    creatorButtonProps = _ref2.creatorButtonProps;
  var _useStyle = useStyle(),
    styles = _useStyle.styles;
  var instance = useSortableList();
  return /*#__PURE__*/_jsxs(Flexbox, {
    className: styles.item,
    direction: 'horizontal',
    align: 'center',
    children: [/*#__PURE__*/_jsx(HandleAction, _objectSpread({
      className: classNames(styles.actions, styles.handle),
      tabIndex: -1,
      cursor: "grab",
      style: {
        width: 14,
        height: 24
      },
      "data-cypress": "draggable-handle"
    }, listeners)), /*#__PURE__*/_jsx(Flexbox, {
      horizontal: true,
      gap: 4,
      width: '100%',
      className: styles.content,
      children: columns.map(function (col, colIndex) {
        var style = {
          flex: 1,
          maxWidth: "".concat(100 / columns.length, "%")
        };
        var props = {
          dataIndex: col.dataIndex,
          value: item[col.dataIndex],
          id: item.id,
          index: index,
          dragging: dragging,
          style: style,
          placeholder: col.placeholder,
          creatorButtonProps: creatorButtonProps
        };
        switch (col.type) {
          default:
          case 'input':
            return /*#__PURE__*/_jsx(ControlInput, _objectSpread({}, props), "".concat(item === null || item === void 0 ? void 0 : item.dataIndex, "-").concat(colIndex));
          case 'select':
            return /*#__PURE__*/_jsx(ControlSelect, _objectSpread({
              options: col.options
            }, props), "".concat(item === null || item === void 0 ? void 0 : item.dataIndex, "-").concat(colIndex));
          case 'custom':
            var Custom = col.render;
            return Custom ? /*#__PURE__*/_jsx(Custom, _objectSpread({
              item: item,
              column: col,
              onChange: function onChange(value) {
                instance.updateItem(_defineProperty({}, col.dataIndex, value), index);
              }
            }, props), "".concat(item === null || item === void 0 ? void 0 : item.dataIndex, "-").concat(colIndex)) : '暂无实现';
        }
      })
    }), /*#__PURE__*/_jsxs(Flexbox, {
      className: classNames(styles.actions, styles.actionsRight),
      direction: 'horizontal',
      align: "center",
      children: [actions, hideRemove ? null : /*#__PURE__*/_jsx(DeleteAction, {
        tabIndex: -1,
        onClick: function onClick() {
          instance.removeItem(index);
        },
        style: {
          height: 22
        }
      })]
    })]
  });
});
export default ColumnItem;
