// @ant-design/pro-editor/es/ColumnList/ColumnList.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["className", "columns", "actions", "hideRemove", "creatorButtonProps"];
import { ConfigProvider, SortableList } from "./..";
import { forwardRef, useCallback } from 'react';
import ColumnItem from "./ColumnItem";
import { Header } from "./Header";
import { useStyle } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
var ColumnList = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var className = _ref.className,
    columns = _ref.columns,
    actions = _ref.actions,
    hideRemove = _ref.hideRemove,
    creatorButtonProps = _ref.creatorButtonProps,
    props = _objectWithoutProperties(_ref, _excluded);
  var _useStyle = useStyle(),
    cx = _useStyle.cx;

  // ColumnList 默认有添加一行按钮
  var customCreatorButtonProps = creatorButtonProps === false ? false : _objectSpread({
    position: 'bottom',
    // 默认生成空数据
    record: function record() {
      return {};
    }
  }, creatorButtonProps);
  var renderItem = useCallback(function (item, _ref2) {
    var index = _ref2.index,
      listeners = _ref2.listeners,
      dragging = _ref2.dragging;
    return /*#__PURE__*/_jsx(ColumnItem, {
      columns: columns,
      item: item,
      dragging: dragging,
      listeners: listeners,
      index: index,
      actions: typeof actions === 'function' ? actions(item, index) : actions,
      creatorButtonProps: customCreatorButtonProps,
      hideRemove: hideRemove
    });
  }, [columns]);
  return /*#__PURE__*/_jsx(ConfigProvider, {
    children: /*#__PURE__*/_jsx(SortableList, _objectSpread({
      ref: ref,
      renderItem: renderItem,
      renderHeader: function renderHeader() {
        return /*#__PURE__*/_jsx(Header, {
          columns: columns
        });
      },
      className: cx(className),
      creatorButtonProps: customCreatorButtonProps
    }, props))
  });
});
export default ColumnList;
