// @ant-design/pro-editor/es/ErrorBoundary/index.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/esm/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
/* eslint-disable react/no-array-index-key */
import { Alert, Button, Empty, Space, Typography, message } from 'antd';
import { Component } from 'react';
import { Flexbox } from 'react-layout-kit';
import { jsxs as _jsxs } from "react/jsx-runtime";
import { jsx as _jsx } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
var Title = Typography.Title,
  Text = Typography.Text;

/******************************************************
 ************************* Dom *************************
 ******************************************************/

/**
 * 错误边界组件属性接口
 */
var ErrorBoundary = /*#__PURE__*/function (_Component) {
  _inherits(ErrorBoundary, _Component);
  var _super = _createSuper(ErrorBoundary);
  function ErrorBoundary() {
    var _this;
    _classCallCheck(this, ErrorBoundary);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super.call.apply(_super, [this].concat(args));
    _defineProperty(_assertThisInitialized(_this), "state", {
      error: undefined,
      info: {
        componentStack: ''
      }
    });
    _defineProperty(_assertThisInitialized(_this), "errorInfo", function () {
      var _info$componentStack$;
      var _this$state = _this.state,
        error = _this$state.error,
        info = _this$state.info;
      return /*#__PURE__*/_jsxs(Flexbox, {
        style: {
          width: 800,
          background: 'rgba(0,0,0,0.02)',
          padding: '24px 40px'
        },
        gap: 12,
        children: [/*#__PURE__*/_jsx("div", {
          children: /*#__PURE__*/_jsxs(Text, {
            children: ["\u9519\u8BEF\u4FE1\u606F\uFF1A", error === null || error === void 0 ? void 0 : error.toString()]
          })
        }), /*#__PURE__*/_jsx(Flexbox, {
          children: (_info$componentStack$ = info === null || info === void 0 ? void 0 : info.componentStack.split('\n').map(function (i, index) {
            return (
              /*#__PURE__*/
              // eslint-disable-next-line react/no-array-index-key
              _jsx("div", {
                style: {
                  textAlign: 'left',
                  fontSize: 10
                },
                children: /*#__PURE__*/_jsx(Text, {
                  type: 'secondary',
                  children: i
                })
              }, i + index)
            );
          })) !== null && _info$componentStack$ !== void 0 ? _info$componentStack$ : null
        })]
      });
    });
    _defineProperty(_assertThisInitialized(_this), "prodInfo", function () {
      var onExportConfig = _this.props.onExportConfig;
      return /*#__PURE__*/_jsxs(Flexbox, {
        align: 'center',
        children: [/*#__PURE__*/_jsx(Empty, {
          image: /*#__PURE__*/_jsx("div", {
            style: {
              fontSize: 80,
              lineHeight: 1
            },
            children: "\uD83E\uDD14"
          }),
          description: /*#__PURE__*/_jsxs(Flexbox, {
            gap: 16,
            children: [/*#__PURE__*/_jsx(Title, {
              level: 5,
              children: " \u7EC4\u4EF6\u6E32\u67D3\u4F3C\u4E4E\u51FA\u4E86\u70B9\u5C0F\u95EE\u9898 "
            }), /*#__PURE__*/_jsxs("div", {
              children: [/*#__PURE__*/_jsx(Text, {
                type: 'secondary',
                children: "\u4E0D\u8FC7\u522B\u62C5\u5FC3\uFF0C\u4F60\u7684\u6570\u636E\u90FD\u8FD8\u5728\uFF0C\u53EF\u4EE5\u5C1D\u8BD5\u5BFC\u51FA\u6570\u636E\u540E\u91CD\u8BD5"
              }), /*#__PURE__*/_jsx("br", {}), /*#__PURE__*/_jsx("div", {
                style: {
                  marginTop: 8
                },
                children: /*#__PURE__*/_jsx(Text, {
                  type: 'secondary',
                  children: "\u5982\u91CD\u8BD5\u540E\u4ECD\u7136\u5B58\u5728\u95EE\u9898\uFF0C\u8BF7\u8054\u7CFB\u5BA2\u670D\u4EBA\u5458"
                })
              })]
            })]
          }),
          children: /*#__PURE__*/_jsx("div", {
            children: /*#__PURE__*/_jsxs(Space, {
              children: [/*#__PURE__*/_jsx(Button, {
                type: 'primary',
                onClick: onExportConfig,
                children: "\u5BFC\u51FA\u914D\u7F6E"
              }), /*#__PURE__*/_jsx(Button, {
                onClick: function onClick() {
                  message.success('已提交Bug');
                },
                children: "\u63D0\u4EA4Bug"
              })]
            })
          })
        }), /*#__PURE__*/_jsx("div", {
          style: {
            marginTop: 24
          },
          children: _this.errorInfo()
        })]
      });
    });
    return _this;
  }
  _createClass(ErrorBoundary, [{
    key: "componentDidCatch",
    value: function componentDidCatch(error, info) {
      this.setState({
        error: error,
        info: info
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
        children = _this$props.children,
        onRetry = _this$props.onRetry,
        _this$props$showDev = _this$props.showDev,
        showDev = _this$props$showDev === void 0 ? process.env.NODE_ENV === 'development' : _this$props$showDev;
      var _this$state2 = this.state,
        error = _this$state2.error,
        info = _this$state2.info;
      if (error) {
        var _info$componentStack$2;
        // You can render any custom fallback UI
        return /*#__PURE__*/_jsx(Flexbox, {
          style: {
            padding: 24
          },
          children: showDev ? /*#__PURE__*/_jsx(Alert, {
            type: "error",
            message: error.toString(),
            showIcon: true,
            description: /*#__PURE__*/_jsxs("div", {
              children: [/*#__PURE__*/_jsx("div", {
                style: {
                  overflow: 'auto',
                  maxHeight: 'calc(100vh - 200px)'
                },
                children: (_info$componentStack$2 = info === null || info === void 0 ? void 0 : info.componentStack.split('\n').map(function (i, index) {
                  return /*#__PURE__*/_jsx("div", {
                    style: {
                      textAlign: 'left',
                      fontSize: 10
                    },
                    children: /*#__PURE__*/_jsx(Text, {
                      type: 'secondary',
                      children: i
                    })
                  }, i + index);
                })) !== null && _info$componentStack$2 !== void 0 ? _info$componentStack$2 : null
              }), /*#__PURE__*/_jsx("div", {
                style: {
                  marginTop: 24
                },
                children: /*#__PURE__*/_jsx(Button, {
                  type: "primary",
                  onClick: onRetry,
                  children: "\u91CD\u8BD5"
                })
              })]
            })
          }) : this.prodInfo()
        });
      }
      return /*#__PURE__*/_jsx(_Fragment, {
        children: children
      });
    }
  }]);
  return ErrorBoundary;
}(Component);
export { ErrorBoundary as default };
