// @ant-design/pro-editor/es/Layout/index.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["header", "footer", "leftPannel", "rightPannel", "bottomPannel", "centerPannel", "type", "themeType"];
import { withProvider } from "../ConfigProvider";
import { HeaderAndFooter } from "./components/HeaderAndFooter";
import { LayoutTypeContainer } from "./components/LayoutTypeContainer";
import { PannelDefault } from "./components/PannelDefault";
import { jsx as _jsx } from "react/jsx-runtime";
var BasicLayout = function BasicLayout(props) {
  var header = props.header,
    footer = props.footer,
    leftPannel = props.leftPannel,
    rightPannel = props.rightPannel,
    bottomPannel = props.bottomPannel,
    centerPannel = props.centerPannel,
    _props$type = props.type,
    type = _props$type === void 0 ? 'Bottom' : _props$type,
    _props$themeType = props.themeType,
    themeType = _props$themeType === void 0 ? 'block' : _props$themeType,
    rest = _objectWithoutProperties(props, _excluded);
  var pannels = [leftPannel, rightPannel, bottomPannel, centerPannel].map(function (props, index) {
    if (props === false) return null;
    return /*#__PURE__*/_jsx(PannelDefault, _objectSpread(_objectSpread({}, props), {}, {
      index: index,
      themeType: (props === null || props === void 0 ? void 0 : props.themeType) || themeType
    }), 'pannel' + index);
  });
  var headerandfooter = [header, footer].map(function (props, index) {
    if (props === false) return null;
    return /*#__PURE__*/_jsx(HeaderAndFooter, _objectSpread(_objectSpread({}, props), {}, {
      type: index === 0 ? 'header' : 'footer',
      themeType: (props === null || props === void 0 ? void 0 : props.themeType) || themeType
    }), index);
  });
  return /*#__PURE__*/_jsx(LayoutTypeContainer, _objectSpread({
    pannels: pannels,
    headerandfooter: headerandfooter,
    themeType: themeType,
    type: type
  }, rest));
};
var WrapperLayout = withProvider(BasicLayout);
export { WrapperLayout as Layout };
