// @ant-design/pro-editor/es/ProBuilder/components/Stage/Empty.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import { Empty } from 'antd';
import { memo } from 'react';
import { useStore } from "../../store";
import { Center } from 'react-layout-kit';
import { shallow } from 'zustand/shallow';
import { useStyle } from "./Empty.style";
import { jsx as _jsx } from "react/jsx-runtime";
var Starter = /*#__PURE__*/memo(function () {
  var _useStore = useStore(function (s) {
      return [s.componentAsset];
    }, shallow),
    _useStore2 = _slicedToArray(_useStore, 1),
    componentAsset = _useStore2[0];
  var _useStyle = useStyle(),
    styles = _useStyle.styles;
  return /*#__PURE__*/_jsx(Center, {
    className: styles.container,
    children: componentAsset.CanvasStarter ? /*#__PURE__*/_jsx(componentAsset.CanvasStarter, {}) : /*#__PURE__*/_jsx(Empty, {
      image: /*#__PURE__*/_jsx("img", {
        src: "https://gw.alipayobjects.com/zos/antfincdn/rxat8ds3j/empty.svg"
      }),
      imageStyle: {
        height: 210,
        marginBottom: 32
      },
      className: styles.default,
      description: '暂无配置信息，请从右侧面板开始编辑'
    })
  });
});
export default Starter;
