// @ant-design/pro-editor/es/ProBuilder/components/Stage/index.js
import { memo } from 'react';
import { Flexbox } from 'react-layout-kit';
import Canvas from "../Canvas";
import Code from "../Code";
import { Controller } from "../Controller";
import Empty from "./Empty";
import { isDesignModeSelector, useStore } from "../../store";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export var Stage = /*#__PURE__*/memo(function (_ref) {
  var hideNavbar = _ref.hideNavbar,
    onCopy = _ref.onCopy;
  var isStarter = useStore(function (x) {
    return x.componentAsset.componentStore(function (s) {
      return x.componentAsset.isStarterMode(s);
    });
  });
  var isDesignMode = useStore(isDesignModeSelector);
  var _useStyles = useStyles(),
    styles = _useStyles.styles;
  return /*#__PURE__*/_jsx("div", {
    className: styles.container,
    style: {
      height: hideNavbar ? '100%' : "calc(100% - 46px)"
    },
    children: /*#__PURE__*/_jsxs(Flexbox, {
      height: '100%',
      children: [isStarter ? /*#__PURE__*/_jsx(Empty, {}) : /*#__PURE__*/_jsx(Canvas, {}), /*#__PURE__*/_jsx(Controller, {}), isDesignMode ? null : /*#__PURE__*/_jsx(Code, {
        onCopy: onCopy
      })]
    })
  });
});
export default Stage;
