// @ant-design/pro-editor/es/ProBuilder/components/Canvas/index.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import { AimOutlined, RedoOutlined, StopOutlined, UndoOutlined } from '@ant-design/icons';
import { Button, Tooltip } from 'antd';
import { memo } from 'react';
import { shallow } from 'zustand/shallow';
import FreeCanvas from "../../../FreeCanvas";
import { useStore } from "../../store";
import Component from "./Component";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var Canvas = /*#__PURE__*/memo(function () {
  var _useStore = useStore(function (s) {
      return [s.editorAwareness.viewport, s.componentAsset, s.enableCanvasInteraction, s.toggleCanvasInteraction, s.internalUpdateEditorAwareness, s.redo, s.undo, s.redoStack().length > 0, s.undoStack().length > 0];
    }, shallow),
    _useStore2 = _slicedToArray(_useStore, 9),
    viewport = _useStore2[0],
    componentAsset = _useStore2[1],
    enableCanvasInteraction = _useStore2[2],
    toggleCanvasInteraction = _useStore2[3],
    updateEditorAwareness = _useStore2[4],
    redo = _useStore2[5],
    undo = _useStore2[6],
    canRedo = _useStore2[7],
    canUndo = _useStore2[8];
  var ErrorBoundary = componentAsset.ErrorBoundary;
  return /*#__PURE__*/_jsx(ErrorBoundary, {
    children: /*#__PURE__*/_jsx(FreeCanvas, {
      viewport: viewport,
      onViewportChange: function onViewportChange(viewport) {
        updateEditorAwareness({
          viewport: viewport
        });
      },
      extraControlBtns: [/*#__PURE__*/_jsx(Tooltip, {
        title: "\u753B\u5E03\u4EA4\u4E92\uFF1A\u5DF2".concat(enableCanvasInteraction ? '启用' : '禁用'),
        children: /*#__PURE__*/_jsx(Button, {
          icon: enableCanvasInteraction ? /*#__PURE__*/_jsx(AimOutlined, {}) : /*#__PURE__*/_jsx(StopOutlined, {}),
          onClick: toggleCanvasInteraction
        })
      }, "interaction"), /*#__PURE__*/_jsxs(Button.Group, {
        children: [/*#__PURE__*/_jsx(Tooltip, {
          title: "\u64A4\u9500",
          children: /*#__PURE__*/_jsx(Button, {
            icon: /*#__PURE__*/_jsx(UndoOutlined, {}),
            onClick: undo,
            disabled: !canUndo
          })
        }), /*#__PURE__*/_jsx(Tooltip, {
          title: "\u91CD\u505A",
          children: /*#__PURE__*/_jsx(Button, {
            icon: /*#__PURE__*/_jsx(RedoOutlined, {}),
            onClick: redo,
            disabled: !canRedo
          })
        })]
      }, 'undo-redo')],
      children: /*#__PURE__*/_jsx(Component, {})
    })
  });
});
export default Canvas;
