// @ant-design/pro-editor/es/types/catogory.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
/**
 * 目录分类
 * 默认包含 四个分类
 * type:  '类型'
 * content: '内容'
 * style: '样式'
 * status:  '状态'
 */

export var CategoryMap = /*#__PURE__*/function (CategoryMap) {
  CategoryMap["type"] = "type";
  CategoryMap["content"] = "content";
  CategoryMap["function"] = "function";
  CategoryMap["style"] = "style";
  CategoryMap["status"] = "status";
  CategoryMap["bind"] = "bind";
  CategoryMap["table"] = "table";
  CategoryMap["customStyle"] = "customStyle";
  return CategoryMap;
}({});
export var configCategoryMap = _defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty({}, CategoryMap.type, {
  title: '类型',
  defaultActive: true
}), CategoryMap.content, {
  title: '内容与功能',
  defaultActive: true
}), CategoryMap.style, {
  title: '样式',
  defaultActive: true
}), CategoryMap.status, {
  title: '状态',
  defaultActive: true
}), CategoryMap.bind, {
  title: '分页字段绑定',
  defaultActive: true
}), CategoryMap.table, {
  title: '表格要素',
  defaultActive: true
}), CategoryMap.customStyle, {
  title: '自定义样式',
  defaultActive: true
});
