// @ant-design/pro-editor/es/ProBuilder/components/NavBar/index.js
import { Space } from 'antd';
import { memo } from 'react';
import { Flexbox } from 'react-layout-kit';
import { useStyle } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var NavBar = /*#__PURE__*/memo(function (_ref) {
  var logo = _ref.logo;
  var _useStyle = useStyle(),
    styles = _useStyle.styles;
  var defaultLogo = /*#__PURE__*/_jsxs(Space, {
    children: [/*#__PURE__*/_jsx("img", {
      src: "https://gw.alipayobjects.com/zos/antfincdn/upvrAjAPQX/Logo_Tech%252520UI.svg",
      alt: "ProBuilder",
      className: styles.img
    }), /*#__PURE__*/_jsx("div", {
      style: {
        fontSize: 16
      },
      children: "ProBuilder"
    })]
  });
  return /*#__PURE__*/_jsx(Flexbox, {
    horizontal: true,
    align: 'center',
    height: 46,
    padding: '0 24px',
    distribution: 'space-between',
    className: styles.container,
    children: /*#__PURE__*/_jsx("div", {
      className: styles.logo,
      children: logo !== null && logo !== void 0 ? logo : defaultLogo
    })
  });
});
export default NavBar;
