// @ant-design/pro-editor/es/ProBuilder/store/slices/config.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import merge from 'lodash.merge';
import { DocWithHistoryManager } from "../../utils/yjs";

// ======== state ======== //

// ======== action ======== //

/**
 * 公共配置操作接口
 */

export var configSlice = function configSlice(set, get) {
  var initialConfigState = {
    // 资产
    componentAsset: null,
    // 文件配置属性
    config: null,
    onConfigChange: null,
    props: {},
    yjsDoc: new DocWithHistoryManager()
  };
  var undoLength = initialConfigState.yjsDoc.undoManager.undoStack.length;
  var redoLength = initialConfigState.yjsDoc.undoManager.redoStack.length;
  return _objectSpread(_objectSpread({}, initialConfigState), {}, {
    undoLength: undoLength,
    redoLength: redoLength,
    resetConfig: function resetConfig() {
      set({
        config: initialConfigState.config,
        props: initialConfigState.props
      });
    },
    /**
     * 内部修改 config 方法
     * 传给 ProTableStore 进行 config 同步
     */
    internalUpdateConfig: function internalUpdateConfig(config, payload, replace) {
      var _get = get(),
        onConfigChange = _get.onConfigChange,
        componentAsset = _get.componentAsset;
      var nextConfig = replace ? config : _objectSpread(_objectSpread({}, get().config), config);
      set({
        config: nextConfig
      }, false, payload);
      onConfigChange === null || onConfigChange === void 0 || onConfigChange({
        config: nextConfig,
        props: componentAsset === null || componentAsset === void 0 ? void 0 : componentAsset.generateProps(nextConfig),
        isEmpty: componentAsset === null || componentAsset === void 0 ? void 0 : componentAsset.isStarterMode(nextConfig)
      });
    },
    exportConfig: function exportConfig() {
      var eleLink = document.createElement('a');
      eleLink.download = 'pro-edior-config.json';
      eleLink.style.display = 'none';
      var blob = new Blob([JSON.stringify(get().config)]);
      eleLink.href = URL.createObjectURL(blob);
      document.body.appendChild(eleLink);
      eleLink.click();
      document.body.removeChild(eleLink);
    },
    setConfig: function setConfig(config) {
      var _ref = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {},
        replace = _ref.replace,
        recordHistory = _ref.recordHistory;
      get().internalUpdateConfig(config, {
        type: '调用 updateConfig 更新',
        payload: config
      }, replace);
      var useAction = merge({}, {
        recordHistory: true
      }, {
        recordHistory: recordHistory
      });
      if (useAction.recordHistory) {
        get().yjsDoc.recordHistoryData({
          config: config
        }, _objectSpread(_objectSpread({}, useAction.payload), {}, {
          timestamp: Date.now()
        }));
      }
    }
  });
};
