// @ant-design/pro-editor/es/ProBuilder/store/slices/configPanel.js
import _typeof from "@babel/runtime/helpers/esm/typeof";
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
export var configPanelSlice = function configPanelSlice(set, get) {
  return {
    /**
     * 更新坐标值
     * @param newPos
     */
    updatePanelPosition: function updatePanelPosition(newPos) {
      var _get = get(),
        editorAwareness = _get.editorAwareness;
      get().internalUpdateEditorAwareness({
        panelPosition: _objectSpread(_objectSpread({}, editorAwareness.panelPosition), newPos)
      });
    },
    updatePanelSize: function updatePanelSize(size) {
      get().internalUpdateEditorAwareness({
        panelSize: size
      });
    },
    /**
     * 触发面板展开折叠
     */
    togglePanelExpand: function togglePanelExpand(nextExpandState) {
      var _get2 = get(),
        internalUpdateEditorAwareness = _get2.internalUpdateEditorAwareness,
        editorAwareness = _get2.editorAwareness;
      if (_typeof(nextExpandState) === undefined) {
        internalUpdateEditorAwareness({
          panelExpand: !editorAwareness.panelExpand
        });
      } else {
        internalUpdateEditorAwareness({
          panelExpand: nextExpandState
        });
      }
    }
  };
};
