// @ant-design/pro-editor/es/ProBuilder/store/slices/general.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
// ======== state ======== //

export var TabKey = /*#__PURE__*/function (TabKey) {
  TabKey[TabKey["canvas"] = 0] = "canvas";
  TabKey[TabKey["code"] = 1] = "code";
  return TabKey;
}({});
export var EditorMode = /*#__PURE__*/function (EditorMode) {
  EditorMode["Design"] = "design";
  EditorMode["Develop"] = "develop";
  return EditorMode;
}({});
var initialGeneralState = {
  // 模式
  mode: EditorMode.Develop
};

// ======== action ======== //

/**
 * 通用公共动作
 */

export var generalSlice = function generalSlice(set, get) {
  return _objectSpread(_objectSpread({}, initialGeneralState), {}, {
    undoStack: function undoStack() {
      return get().yjsDoc.undoManager.undoStack;
    },
    redoStack: function redoStack() {
      return get().yjsDoc.undoManager.redoStack;
    },
    undo: function undo() {
      var _get = get(),
        yjsDoc = _get.yjsDoc,
        internalUpdateConfig = _get.internalUpdateConfig;
      var stack = yjsDoc.undo();
      var _yjsDoc$getHistoryJSO = yjsDoc.getHistoryJSON(),
        config = _yjsDoc$getHistoryJSO.config;
      internalUpdateConfig(config, {
        type: 'history/undo',
        payload: stack
      }, true);
    },
    redo: function redo() {
      var _get2 = get(),
        yjsDoc = _get2.yjsDoc,
        internalUpdateConfig = _get2.internalUpdateConfig;
      var stack = yjsDoc.redo();
      var _yjsDoc$getHistoryJSO2 = yjsDoc.getHistoryJSON(),
        config = _yjsDoc$getHistoryJSO2.config;
      internalUpdateConfig(config, {
        type: 'history/redo',
        payload: stack
      }, true);
    }
  });
};
