// @ant-design/pro-editor/es/ComponentAsset/store/createAssetStore.js
import isEqual from 'fast-deep-equal';
import { createContext, optionalDevtools } from 'zustand-utils';
import { createWithEqualityFn } from 'zustand/traditional';
var _createContext = createContext(),
  Provider = _createContext.Provider,
  useStore = _createContext.useStore,
  useStoreApi = _createContext.useStoreApi;
export var createAssetStore = function createAssetStore(createStore, options) {
  var store = function store() {
    var devtoolsOptions = (options === null || options === void 0 ? void 0 : options.devtools) === false ? {} : (options === null || options === void 0 ? void 0 : options.devtools) === true ? {
      name: 'ProEditor-AssetStore'
    } : options === null || options === void 0 ? void 0 : options.devtools;
    var devtools = optionalDevtools(!((options === null || options === void 0 ? void 0 : options.devtools) === false));
    return createWithEqualityFn()(devtools(createStore, devtoolsOptions), isEqual);
  };
  return {
    Provider: Provider,
    createStore: store,
    useStoreApi: useStoreApi
  };
};
export var createUseAssetStore = function createUseAssetStore() {
  return {
    useStore: useStore,
    useStoreApi: useStoreApi,
    AssetProvider: Provider
  };
};
