// @ant-design/pro-editor/es/ComponentAsset/store/createTestAssetStore.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import isEqual from 'fast-deep-equal';
import { createWithEqualityFn } from 'zustand/traditional';
/**
 * 创建测试用的 AssetStore，请不要在生产环境使用
 * @param createStore
 */
export var createTestAssetStore = function createTestAssetStore(createStore) {
  var useEditorStore = createWithEqualityFn(function (set, get) {
    return {
      config: null,
      setConfig: function setConfig(config) {
        set({
          config: _objectSpread(_objectSpread({}, get().config), config)
        });
      },
      getConfig: function getConfig() {
        return get().config;
      }
    };
  }, isEqual);
  var useAssetStore = createWithEqualityFn(createStore, isEqual);
  var _useEditorStore$getSt = useEditorStore.getState(),
    getConfig = _useEditorStore$getSt.getConfig,
    setConfig = _useEditorStore$getSt.setConfig;
  var init = function init() {
    useAssetStore.setState({
      getConfig: getConfig,
      setConfig: setConfig
    });
    useEditorStore.subscribe(function (_ref) {
      var config = _ref.config;
      useAssetStore.setState(config);
    });
  };
  return {
    useAssetStore: useAssetStore,
    init: init,
    useEditorStore: useEditorStore
  };
};
