// @ant-design/pro-editor/es/ProBuilder/components/Code/index.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import { DownOutlined, RightOutlined } from '@ant-design/icons';
import { DraggablePanel, Highlight } from "../../..";
import { useTheme } from 'antd-style';
import { memo, useState } from 'react';
import { Flexbox } from 'react-layout-kit';
import { shallow } from 'zustand/shallow';
import { useStore } from "../../store";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var HEADER_HEIGHT = 36;
/**
 * 用于组装成品代码并高亮展示
 */
export var CodePanel = /*#__PURE__*/memo(function (props) {
  var _useStore = useStore(function (s) {
      return [s.config, s.componentAsset];
    }, shallow),
    _useStore2 = _slicedToArray(_useStore, 2),
    config = _useStore2[0],
    componentAsset = _useStore2[1];
  var _useStyles = useStyles(),
    styles = _useStyles.styles,
    cx = _useStyles.cx;
  var _useState = useState(true),
    _useState2 = _slicedToArray(_useState, 2),
    expand = _useState2[0],
    setExpand = _useState2[1];
  var _useState3 = useState(220),
    _useState4 = _slicedToArray(_useState3, 2),
    height = _useState4[0],
    setHeight = _useState4[1];
  var onCopy = props.onCopy;
  var _useTheme = useTheme(),
    isDarkMode = _useTheme.isDarkMode;
  var configCode = componentAsset.generateCode(config);
  return /*#__PURE__*/_jsxs(DraggablePanel, {
    onSizeChange: function onSizeChange(delta) {
      setHeight(height + delta.height);
    },
    placement: 'bottom',
    size: {
      height: expand ? height : HEADER_HEIGHT
    },
    minHeight: HEADER_HEIGHT,
    className: styles.container,
    expandable: false,
    children: [/*#__PURE__*/_jsxs(Flexbox, {
      padding: '6px 8px',
      horizontal: true,
      distribution: 'space-between',
      align: 'center',
      className: cx(styles.header, !expand ? styles.collapse : ''),
      onClick: function onClick() {
        return setExpand(!expand);
      },
      children: [/*#__PURE__*/_jsxs(Flexbox, {
        horizontal: true,
        gap: 12,
        className: styles.headerTitle,
        children: [expand ? /*#__PURE__*/_jsx(DownOutlined, {}) : /*#__PURE__*/_jsx(RightOutlined, {}), "\u4EE3\u7801\u751F\u6210"]
      }), /*#__PURE__*/_jsx(Flexbox, {
        horizontal: true,
        gap: 8,
        onClick: function onClick(e) {
          e.stopPropagation();
        }
      })]
    }), /*#__PURE__*/_jsx(Flexbox, {
      className: styles.code,
      children: /*#__PURE__*/_jsx(Highlight, {
        theme: isDarkMode ? 'dark' : 'light',
        lineNumber: true,
        language: "tsx",
        onCopy: onCopy,
        showLanguage: false,
        style: {
          height: '100%',
          overflow: 'auto'
        },
        children: configCode
      })
    })]
  });
});
export default CodePanel;
