// @ant-design/pro-editor/es/ProBuilder/components/Code/style.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/esm/taggedTemplateLiteral";
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5;
import { createStyles } from "../../../theme";
export var useStyles = createStyles(function (_ref) {
  var token = _ref.token,
    stylish = _ref.stylish,
    css = _ref.css;
  return {
    container: css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n      border-top: 1px solid ", ";\n      display: flex;\n      flex-direction: column;\n    "])), token.colorBorder),
    header: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n      background: ", ";\n      cursor: pointer;\n    "])), token.colorBgContainer),
    collapse: css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n      background: ", ";\n      &:hover {\n        background: ", ";\n      }\n    "])), token.colorFillQuaternary, token.colorFillTertiary),
    headerTitle: css(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n      ", "\n      ", ";\n\n      padding: 2px 8px;\n\n      color: ", ";\n      border-radius: ", "px;\n    "])), stylish.containerBgHover, stylish.textInfo, token.colorTextSecondary, token.borderRadius),
    code: css(_templateObject5 || (_templateObject5 = _taggedTemplateLiteral(["\n      background: ", ";\n      height: 100%;\n    "])), token.colorFillQuaternary)
  };
});
