// @ant-design/pro-editor/es/ProBuilder/container/index.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["style", "__EDITOR_STORE_DEVTOOLS__", "editorRef"];
import { ConfigProvider } from "../..";
import { App } from 'antd';
import { memo } from 'react';
import { HotkeysProvider } from 'react-hotkeys-hook';
import Content from "./App";
import Provider from "./Provider";
import StoreUpdater from "./StoreUpdater";
import { useStyle } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export var ProBuilder = /*#__PURE__*/memo(function (props) {
  var style = props.style,
    __EDITOR_STORE_DEVTOOLS__ = props.__EDITOR_STORE_DEVTOOLS__,
    editorRef = props.editorRef,
    res = _objectWithoutProperties(props, _excluded);
  var _useStyle = useStyle(),
    styles = _useStyle.styles;
  return /*#__PURE__*/_jsx(HotkeysProvider, {
    initiallyActiveScopes: ['pro-builder'],
    children: /*#__PURE__*/_jsxs(Provider, {
      devtoolOptions: __EDITOR_STORE_DEVTOOLS__,
      children: [/*#__PURE__*/_jsx(ConfigProvider, {
        children: /*#__PURE__*/_jsx(App, {
          className: styles.app,
          style: style,
          children: /*#__PURE__*/_jsx(Content, _objectSpread({}, res))
        })
      }), /*#__PURE__*/_jsx(StoreUpdater, _objectSpread({
        editorRef: editorRef
      }, res))]
    })
  });
});
export { ProBuilderProvider } from "./Provider";
