// @ant-design/pro-editor/es/ProEditor/hooks/useProEditor.js
import { useMemoizedFn } from 'ahooks';
import { useMemo } from 'react';
import { useStoreApi } from "../store";

/**
 * ProBuilder 实例对象
 * @template Config - 配置信息的类型
 * @template Props - 组件属性的类型
 */

export var useProEditor = function useProEditor() {
  var storeApi = useStoreApi();
  var _storeApi$getState = storeApi.getState(),
    undoStack = _storeApi$getState.undoStack,
    undoLength = _storeApi$getState.undoLength,
    redoLength = _storeApi$getState.redoLength,
    redoStack = _storeApi$getState.redoStack,
    setConfig = _storeApi$getState.setConfig,
    exportConfig = _storeApi$getState.exportConfig,
    resetConfig = _storeApi$getState.resetConfig,
    undo = _storeApi$getState.undo,
    redo = _storeApi$getState.redo;
  var getConfig = useMemoizedFn(function () {
    return storeApi.getState().config;
  });
  var getProps = useMemoizedFn(function () {
    return storeApi.getState().props;
  });
  return useMemo(function () {
    return {
      getConfig: getConfig,
      setConfig: setConfig,
      exportConfig: exportConfig,
      resetConfig: resetConfig,
      undo: undo,
      redo: redo,
      undoStack: undoStack,
      redoStack: redoStack,
      undoLength: undoLength,
      redoLength: redoLength,
      getProps: getProps
    };
  }, []);
};
