// @ant-design/pro-editor/es/SortableList/hooks/useSortableList.js
import { useMemoizedFn } from 'ahooks';
import { useStoreApi } from "../store";

/**
 * SortableList 实例对象
 * @template T 节点数据类型
 */

export var useSortableList = function useSortableList() {
  var storeApi = useStoreApi();
  var getActiveId = useMemoizedFn(function () {
    return storeApi.getState().activeId;
  });
  var getIdByIndex = useMemoizedFn(function (index) {
    var _storeApi$getState = storeApi.getState(),
      keyManager = _storeApi$getState.keyManager;
    var indexId = keyManager[index] || null;
    return indexId;
  });
  var getValue = useMemoizedFn(function () {
    return storeApi.getState().value;
  });
  var addItem = function addItem(item, index) {
    return storeApi.getState().dispatchListData({
      type: 'addItem',
      item: item,
      index: index
    });
  };
  var removeItem = function removeItem(index) {
    return storeApi.getState().dispatchListData({
      type: 'removeItem',
      index: index
    });
  };
  var updateItem = function updateItem(item, index) {
    return storeApi.getState().dispatchListData({
      type: 'updateItem',
      item: item,
      index: index
    });
  };
  return {
    getActiveId: getActiveId,
    getValue: getValue,
    addItem: addItem,
    removeItem: removeItem,
    getIdByIndex: getIdByIndex,
    updateItem: updateItem
  };
};
